src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -1977,6 +1976,14 @@ impl Step for Assemble {
             }
         }
 
+        {
+            // `llvm-strip` is used by rustc, which is actually just a symlink to `llvm-objcopy`,
+            // so copy and rename `llvm-objcopy`.
+            let src_exe = exe("llvm-objcopy", target_compiler.host);
+            let dst_exe = exe("rust-objcopy", target_compiler.host);
+            builder.copy_link(&libdir_bin.join(src_exe), &libdir_bin.join(dst_exe));
+        }
```

Comment:
This breaks bootstrapping when the bootstrap compiler doesn't contain any LLVM tools as is the case when cg_clif bootstraps rustc:

```
Building stage0 tool lld-wrapper (x86_64-unknown-linux-gnu)
     Compiling lld-wrapper v0.1.0 (/home/runner/work/rustc_codegen_cranelift/rustc_codegen_cranelift/rust/src/tools/lld-wrapper)
      Finished `release` profile [optimized] target(s) in 0.43s
  thread 'main' panicked at src/lib.rs:1714:24:
src.symlink_metadata() failed with No such file or directory (os error 2) ("src = /home/runner/work/rustc_codegen_cranelift/rustc_codegen_cranelift/rust/build/x86_64-unknown-linux-gnu/stage1/lib/rustlib/x86_64-unknown-linux-gnu/bin/llvm-objcopy")
```

It should probably check `builder.config.llvm_enabled(target_compiler.host) && builder.config.llvm_tools_enabled` before copying to ensure that `llvm-objcopy` exists in the first place.

Reply:
Thanks for the heads-up, I'll send a follow-up to fix that.

Reply:
Follow-up: https://github.com/rust-lang/rust/pull/132720